src/UI/Component/Input/Factory.php

```diff
@@ -0,0 +1,46 @@
+<?php
+namespace ILIAS\UI\Component\Input;
+/**
+ * This is what a factory for inputs looks like.
+ */
+interface Factory {
+
+	/**
+	 * ---
+	 * description:
+	 *   purpose: >
+	 *     Ratings allow the user to pick a number from a fix scale of five to express valuation.
+	 *     Each (full-number) position on the scale has a term assigned that "translates"
+	 *     the numerical value into a colloquial term.
+	 *   composition: >
+	 *     A rating-input consists of five identical graphical elements that can be selected/clicked
+	 *     and a string that is displayed dependent on the selected item.
+	 *     Additionally, a title plus a short phrase is shown explaining the topic to be rated.
+	 *   effect: >
+	 *     A RatingSelector-input displays five stars in a horizontal row.
+	 *     By selecting one position, the star at that position is highlighted as well as all stars left to it,
+	 *     such rather expressing an amount than a position.
+	 *     A term specifying/explaining the selected value is displayed.
+	 *     The probably most famous rating input is the amazon five-star rating.
```

Comment:
Attention: Currently in ILIAS, this rating is not limited to making a rating (performing an input) but also showing the current average of a rating. See: [Pipeline Document](https://docs.google.com/document/d/1JoInU0-tkRHDOMjMTNziV-hJb6ypDCI1zT9OGtAThZk/edit#heading=h.rww0pri4fw8u) for a summary of existing ratings.

The composition should allow the following core functionality:

- Performing a rating (check)
- Showing your current rating (check)
- Removing the own rating, setting the status to not set (not checked)
- Showing the current average ratings of all users that may have performed a rating (not checked)